<launch>
    <arg name="robot_name" default="robot1"/>
    <!-- add SLAM node -->
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
        <param name="base_frame" value="$(arg robot_name)/base_footprint"/>
        <param name="odom_frame" value="$(arg robot_name)/odom"/>
        <param name="map_frame" value="map"/>
        <param name="map_update_interval" value="1.0"/>
        <param name="maxUrange" value="5.0"/>
        <param name="sigma" value="0.05"/>
        <param name="kernelSize" value="1"/>
        <param name="lstep" value="0.05"/>
        <param name="astep" value="0.05"/>
        <param name="iterations" value="5"/>
        <param name="lsigma" value="0.075"/>
        <param name="ogain" value="3.0"/>
        <param name="minimumScore" value="50"/>
    </node>

    <!-- Basic Navigation -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <rosparam file="$(find cs_bot)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find cs_bot)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find cs_bot)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find cs_bot)/config/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find cs_bot)/config/move_base_params.yaml" command="load" />

        <!-- explicitly specify frame_id -->
        <param name="global_costmap/global_frame" value="map"/>
        <param name="global_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint"/>
        <param name="local_costmap/global_frame" value="$(arg robot_name)/odom"/>
        <param name="local_costmap/robot_base_frame" value="$(arg robot_name)/base_footprint"/>
        <param name="planner_frequency" value="1.0"/>
        <param name="planner_patience" value="5.0"/>
        <param name="controller_frequency" value="5.0"/>
        <param name="controller_patience" value="15.0"/>
        <param name="recovery_behavior_enabled" value="true"/>
        <param name="clearing_rotation_allowed" value="true"/>
    </node>
</launch>
